waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/PrefixMapping.java

```diff
@@ -15,32 +15,76 @@
  */
 package com.hotels.bdp.waggledance.mapping.model;
 
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.CAT_NAME;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.DB_NAME;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.parseDbName;
+import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.prependNotNullCatToDbName;
+
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.hadoop.hive.metastore.api.MetaException;
+
 public class PrefixMapping extends MetaStoreMappingDecorator {
 
   public PrefixMapping(MetaStoreMapping metaStoreMapping) {
     super(metaStoreMapping);
   }
 
   @Override
-  public String transformOutboundDatabaseName(String databaseName) {
-    return getDatabasePrefix() + super.transformOutboundDatabaseName(databaseName);
+  public String transformOutboundDatabaseName(String databaseName)  {
+      if(hasCatalogName(databaseName)) {
+        String[] catalogAndDatabase = new String[0];
+        try {
+          catalogAndDatabase = parseDbName(databaseName, null);
+        }
+        catch (MetaException e) {
+          //FIXME
+          throw new RuntimeException(e);
+        }
+        String dbName = catalogAndDatabase[DB_NAME];
+        String catalogName = catalogAndDatabase[CAT_NAME];
+        String prefixedDbName = getDatabasePrefix() + super.transformOutboundDatabaseName(dbName);
+        return prependNotNullCatToDbName(catalogName, prefixedDbName);
+      }else {
+        return getDatabasePrefix() + super.transformOutboundDatabaseName(databaseName);
+      }
+  }
+
+  private static boolean hasCatalogName(String dbName) {
```

Comment:
Maybe worth moving to a `CatalogUtils` class of some form? There might be some other duplicated/common catalog related code we could put there...